lib/active_merchant/billing/gateways/pay_trace.rb

```diff
@@ -43,23 +43,19 @@ def initialize(options = {})
       end
 
       def purchase(money, payment_or_customer_id, options = {})
-        post = {}
-        add_amount(post, money, options)
-        if customer_id?(payment_or_customer_id)
-          post[:customer_id] = payment_or_customer_id
-          endpoint = ENDPOINTS[:customer_id_sale]
+        if visa_or_mastercard?(options)
+          MultiResponse.run(:use_first_response) do |r|
+            customer_id?(payment_or_customer_id) ? endpoint = [ENDPOINTS][:customer_id_sale] : endpoint = ENDPOINTS[:keyed_sale]
```

Comment:
We can (I think) tidy this up by putting `endpoint =` at the beginning like so: 

```
endpoint = customer_id?(payment_or_customer_id) ? [ENDPOINTS][:customer_id_sale] : ENDPOINTS[:keyed_sale]
```

Reply:
Revised! Thanks for teaching me a thing with that!